core/src/main/java/ai/timefold/solver/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -417,6 +407,29 @@ private void processSolutionAnnotations(DescriptorPolicy descriptorPolicy) {
                 new LookUpStrategyResolver(descriptorPolicy, lookUpStrategyType);
     }
 
+    private @NonNull PlanningSolution extractTopMostPlanningSolutionAnnotation() {
```

Comment:
We're returning the child class first, not the parent class. The method's name suggests that we always return the parent first.